src/main/java/org/mule/extension/sftp/internal/connection/SftpFileSystem.java

```diff
@@ -52,30 +53,31 @@
  */
 public class SftpFileSystem extends AbstractExternalFileSystem {
 
-  protected final SftpClient client;
-  protected final CopyCommand copyCommand;
-  protected final CreateDirectoryCommand createDirectoryCommand;
-  protected final DeleteCommand deleteCommand;
-  protected final SftpListCommand listCommand;
-  protected final MoveCommand moveCommand;
-  protected final SftpReadCommand readCommand;
-  protected final RenameCommand renameCommand;
-  protected final WriteCommand writeCommand;
-  private final LockFactory lockFactory;
-
+  public static final String ROOT = "/";
 
   private static String resolveBasePath(String basePath, SftpClient client) {
     if (isBlank(basePath)) {
       try {
-        return client.getWorkingDirectory();
+        basePath = client.getWorkingDirectory();
       } catch (Exception e) {
         throw new IllegalPathException("SFTP working dir was not specified and failed to resolve a default one",
                                        e);
       }
     }
-    return basePath;
+    return createUri(ROOT, basePath).getPath();
   }
 
+  protected final SftpClient client;
```

Comment:
Is there any reason to move them down here? If not, I would move them back up.

Reply:
Consistency with FTP 😄